### PR TITLE
Properly type Promises

### PR DESCRIPTION
**Pull request by AnnikaCodes**
This fixes several things that TypeScript has been complaining about lately.

### CODE REVIEW

File: server/ip-tools.ts [hunk 1]
```diff
@@ -456,7 +456,7 @@ export const IPTools = new class {
 
 			const ipNumber = IPTools.ipToNumber(ip);
 			for (const range of IPTools.ranges) {
-				if (ipNumber >= range.minIP && ipNumber <= range.maxIP) {
+				if (ipNumber >= range.minIP && ipNumber <= range.maxIP && range.host) {
```
Comment (Zarel) [inline server/ip-tools.ts:459]:
```suggestion
				if (ipNumber >= range.minIP && ipNumber <= range.maxIP) {
```

File: server/ip-tools.ts [hunk 2]
```diff
@@ -469,7 +469,7 @@ export const IPTools = new class {
 				if (!hosts || !hosts[0]) {
 					if (ip.startsWith('50.')) {
 						resolve('comcast.net?/res');
-					} else if (ipNumber >= telstraRange.minIP && ipNumber <= telstraRange.maxIP) {
+					} else if (ipNumber >= telstraRange.minIP && ipNumber <= telstraRange.maxIP && telstraRange.host) {
```
Comment (Zarel) [inline server/ip-tools.ts:472]:
```suggestion
					} else if (ipNumber >= telstraRange.minIP && ipNumber <= telstraRange.maxIP) {
```

Comment (AnnikaCodes) [reply server/ip-tools.ts]: oh i forgot to remove that lol

**Comment by Zarel** [comment]
ty!

**Pull request merged**